sealer/cmd/join.go

```diff
@@ -0,0 +1,96 @@
+// Copyright © 2021 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/alibaba/sealer/cert"
+
+	"github.com/alibaba/sealer/apply"
+	"github.com/alibaba/sealer/common"
+	"github.com/alibaba/sealer/logger"
+	"github.com/spf13/cobra"
+)
+
+var clusterName string
+var joinArgs *common.RunArgs
+
+func getClusterName(sealerPath string) []string {
+	files, err := ioutil.ReadDir(sealerPath)
+	if err != nil {
+		logger.Error(err)
+		os.Exit(1)
```

Comment:
Please do not directly in an inner function. You could return an error, and make the function caller to decide which way to choose, exit or throw an error.

Reply:
OK